xline/src/server/kv_server.rs

```diff
@@ -494,19 +560,26 @@ where
         &self,
         request: tonic::Request<CompactionRequest>,
     ) -> Result<tonic::Response<CompactionResponse>, tonic::Status> {
-        let compact_req = request.get_ref();
-        debug!("Receive grpc request: {:?}", compact_req);
-        Err(tonic::Status::new(
-            tonic::Code::Unimplemented,
-            "Not Implemented".to_owned(),
-        ))
+        debug!("Receive CompactionRequest {:?}", request);
+        let compacted_revision = self.kv_storage.compacted_revision();
+        let current_revision = self.kv_storage.revision();
+        let req = request.get_ref();
+        Self::check_compact_request(req, compacted_revision, current_revision)?;
```

Comment:
The validation check should be moved to `kv_store`, otherwise the outside client will have no way to check if the request is valid.

Reply:
I don't get the point. Check what? And why should an outside client check the request?

Reply:
I'll leave another issue to track and fix it in another pr.

Reply:
FYI: #384